### ISSUE TITLE
[CP] iOS screen flickering speculative fix and logging

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#114174]
### issue_link

https://github.com/flutter/flutter/issues/100522

### Commit Hash

n/a

### Target

stable

### pr_link

n/a

### Impacted Users

All iOS apps

### Impact Description

This CP request is a speculative fix (and additional logging) for the iOS screen flickering issue linked above. One possibility is that the root cause of that issue is a race in Metal command buffer submission when an app is going into the background. The two commits below have been on beta and master channel for a good while. Since the next stable is still a way off, we'd like to alleviate some pain in the meantime in case these changes help fix or diagnose the issue.

https://skia-review.googlesource.com/c/skia/+/575667 <- We suspect that this may fix a race in command buffer submission when going into the background.

https://skia-review.googlesource.com/c/skia/+/566883 <- If that change does not fix the flickering issue, then this change will give more verbose logs when command buffers fail.

The 3.3 release used Skia at https://github.com/google/skia/commit/e2ba2f5b20e0d188dbfb5d3df64aa9787ebb3f37. We'll need to CP the above patches onto a new Skia branch based on that hash.

### Workaround

There is no workaround

### Risk

medium

### Test Coverage

yes

### Validation Steps

We don't have a local reproduction for this issue, so the fix is speculative.

**Comment by CaseyHillers** [comment]
Thank you @christopherfujino for the background on skia CPs. The release team doesn't have a playbook on this, so I'm going to assign to me to figure out and create a playbook entry.

I'm currently blocked from creating new skia branches. I've gone ahead and filed https://bugs.chromium.org/p/skia/issues/detail?id=13865 to request access for the release team.

**Comment by CaseyHillers** [comment]
https://skia.googlesource.com/skia/+/refs/heads/flutter/flutter-3.2-candidate.5 is created

**Comment by CaseyHillers** [comment]
I sent the CP CLs:

https://skia-review.googlesource.com/c/skia/+/597196

https://skia-review.googlesource.com/c/skia/+/597197

**Comment by zanderso** [comment]
I don't have +2 in the Skia repo, so we'll need a review from folks on the Skia team on those CLs.

**Comment by CaseyHillers** [comment]
Marking approved on behalf of @chinmaygarde (based on the lgtms in the linked CLs)

**Comment by CaseyHillers** [comment]
This was released in 3.3.7

**Comment by lonyee1989** [comment]
My app has a lot of animations, so it is easy to switch back and forth to the background many times. The current test 3.3.7 still hasn't fixed it.

It work well in debug mode, but it will flickering work in release mode.

@CaseyHillers @zanderso 

```
2022-11-07 14:36:36.294111+0800 Runner[4777:2920461] Metal GPU Frame Capture Enabled
2022-11-07 14:36:36.294240+0800 Runner[4777:2920461] Metal API Validation Enabled
[2022-11-07 14:36:36.439] [0] [debug] /tmp/jenkins/IRIS-SDK/base/internal/log/iris_logger.cc:35 InitLogger /var/mobile/Containers/Data/Application/127F47F5-61A9-4499-AC94-3FFAA77AF131/Library/Caches/
2022-11-07 14:36:36.481449+0800 Runner[4777:2920461] Unable to upload statistics，appkey or stats is nil
2022-11-07 14:36:36.481471+0800 Runner[4777:2920461] device token changed, reconnect
2022-11-07 14:36:36.954976+0800 Runner[4777:2920461] [RC-Flutter-IM] iOS setReconnectKickEnable, start param:1
2022-11-07 14:36:36.956640+0800 Runner[4777:2920461] RCFwLog, LogMode = 0, monitorLevel = 5, monitorType = 268434431
2022-11-07 14:36:37.073836+0800 Runner[4777:2920800] flutter: 2 has been spawned
2022-11-07 14:36:37.075708+0800 Runner[4777:2920800] flutter: initialized
2022-11-07 14:36:37.397608+0800 Runner[4777:2920461] [RC-Flutter-IM] iOS onConnectionStatusChanged,status:10
2022-11-07 14:36:37.410222+0800 Runner[4777:2920461] clientIp:
[000:000] [83203] (messagequeue.cc:523): Message took 78ms to dispatch. Posted from: Connect@/Users/jenkins/slave/workspace/iOS-SDK-Release/protocol-cpp/rcps/rcps_client_impl.cpp:546
2022-11-07 14:36:37.488674+0800 Runner[4777:2920793] [RC-Flutter-IM] iOS connect, dbOpened，code: 0

2022-11-07 14:36:37.500244+0800 Runner[4777:2920800] [VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method getRCNotification on channel com.push.msg.method)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:294)
<asynchronous suspension>
#1      _IndexState.initNotification (package:xxxx/pages/index.dart:171)
<asynchronous suspension>
2022-11-07 14:36:37.515019+0800 Runner[4777:2920785] L-collection-S,host:39.105.152.201,port:443,cost:26,nettype:1,error:0
2022-11-07 14:36:37.515146+0800 Runner[4777:2920793] L-collection-S,host:39.105.152.201,port:8000,cost:0,nettype:1,error:-1
2022-11-07 14:36:37.515326+0800 Runner[4777:2920790] L-collection-S,host:t-tcpproxy-cn.ronghub.com,port:443,cost:0,nettype:1,error:-1
2022-11-07 14:36:37.515342+0800 Runner[4777:2920785] L-collection-S,host:t-tcpproxy-cn.ronghub.com,port:8085,cost:0,nettype:1,error:-1
2022-11-07 14:36:37.515615+0800 Runner[4777:2920793] [RC:IM-stats_cmp-S]data:{"hos":"http:\/\/39.105.152.201:443","fg":1,"spr":0,"dt":0,"pr":0,"bid":"xxxxxx.music","idx":0,"net":"wifi","rct":0,"dip":"","cpd":"62BD596F2C9857C993DBCC2F2DDBE47D","ddu":0,"dh":["192.168.0.1"],"ptc":2,"cod":0,"cr":[],"dur":26}
[000:064] [83203] (rcps_pull_message.cpp:55): PullMessageRequest: P-reason-C;;;recv;;;1667561924461, send: 1666864666149, broadcast: 0
2022-11-07 14:36:37.553327+0800 Runner[4777:2920461] [RC-Flutter-IM] iOS onConnectionStatusChanged,status:0
2022-11-07 14:36:37.553685+0800 Runner[4777:2920790] [RC-Flutter-IM] iOS connect, success
[000:173] [83203] (rcps_pull_message.cpp:98): DealWithResponse: P-reason-C;;;sync;;;1667750400000,broadcast: 0,finished: 1,size: 0
2022-11-07 14:36:38.306210+0800 Runner[4777:2920790] [NSURLSessionTaskMetrics] 0.411221
2022-11-07 14:36:38.307846+0800 Runner[4777:2920795] [RC:IM-stats_navi-S]data:{"hos":"https:\/\/sirb.cn.rongnav.com\/navi.json","fg":1,"spr":0,"dt":1,"pr":0,"bid":"xxxxxx.music","net":"wifi","dip":"1.117.137.95","ddu":0,"dh":["192.168.0.1"],"ptc":1,"cod":200,"cr":[],"dur":411}

[180:066] [83203] (rcps_pull_message.cpp:55): PullMessageRequest: P-reason-C;;;recv;;;1667561924461, send: 1666864666149, broadcast: 0
[180:535] [83203] (rcps_pull_message.cpp:98): DealWithResponse: P-reason-C;;;sync;;;1667750400000,broadcast: 0,finished: 1,size: 0
```
